pkg/cmd/roachtest/roachtestutil/validation_check.go

```diff
@@ -46,20 +58,40 @@ WHERE t.status NOT IN ('RANGE_CONSISTENT', 'RANGE_INDETERMINATE')`)
 		return nil
 	}
 	defer rows.Close()
+
+	logEvery := log.Every(time.Minute)
+	logEvery.ShouldLog() // don't immediately log
 	var finalErr error
+	var i int
 	for rows.Next() {
 		var rangeID int32
 		var prettyKey, status, detail string
 		if scanErr := rows.Scan(&rangeID, &prettyKey, &status, &detail); scanErr != nil {
 			l.Printf("consistency check failed with %v; ignoring", scanErr)
-			return nil
+			return finalErr // return partial finalErr anyway
+		}
+		// Only detect replica inconsistencies, and ignore MVCC stats mismatches
+		// since these can happen in rare cases due to lease requests not respecting
+		// latches: https://github.com/cockroachdb/cockroach/issues/93896
+		//
+		// TODO(erikgrinaker): We should take storage checkpoints for inconsistent
+		// ranges as well. This requires support in check_consistency() such that we
+		// take the checkpoints at the same Raft log index across nodes.
+		if status == kvpb.CheckConsistencyResponse_RANGE_INCONSISTENT.String() {
+			finalErr = errors.CombineErrors(finalErr,
+				errors.Newf("r%d (%s) is inconsistent: %s %s\n", rangeID, prettyKey, status, detail))
```

Comment:
In the worst case when thousands of ranges are inconsistent, I don't think it will be a good idea to `CombineErrors` like this, especially since each `errors.Newf` is capturing a stack trace.

Reply:
True, and when we add checkpointing we'll definitely not want to checkpoint too many ranges. I'll add a limit here.

Reply:
Limited the error to 10 ranges, but added logging of all inconsistencies.